serde/src/de/mod.rs

```diff
@@ -1,6 +1,12 @@
 //! Generic deserialization framework.
 
+#[cfg(feature = "std")]
 use std::error;
+#[cfg(not(feature = "std"))]
+use error;
```

Comment:
maybe we should patch libcore/libstd to move the error module to libcore and reexport in libstd?